File: backend-intermediate-cs/OrderManagement/Models/OrderDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace OrderManagement.Models {
    [Table("order_detail")]
    public class OrderDetail
    {
        [Key]
        [Column("detail_id")]
        public int DetailId { get; set; }

        [Column("amount")]
        public double Amount { get; set; }

        [Required]
        [Column("quantity")]
        [Range(1, int.MaxValue, ErrorMessage = "Quantity must be at least 1")]
        public int Quantity { get; set; }

        [Required]
        [Column("order_number")]
        public int OrderNumber { get; set; }

        [Required]
        [Column("product_id")]
        public int ProductId { get; set; }
        [JsonIgnore]
        // Navigation properties
        public Order? Order { get; set; }
        
        public Product? Product { get; set; }
    }
}

File: qa-fundamentals/OrderManagementJava/src/main/java/org/orders/model/OrderDetail.java
package org.orders.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "order_detail")
@Getter
@Setter
public class OrderDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "detail_id")
    private Integer detailId;

    @Column(name = "amount")
    private Double amount;

    @NotNull
    @Min(value = 1, message = "Quantity must be at least 1")
    @Column(name = "quantity", nullable = false)
    private Integer quantity;


    @NotNull
    @Column(name = "product_id", nullable = false)
    private Integer productId;

    @JsonIgnore
    // Navigation properties
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_number")
    private Order order;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", insertable = false, updatable = false)
    private Product product;
}
